### PR TITLE
BUG: Remove invalid read in searchsorted if needle is empty

### PR DESCRIPTION
**Pull request by seberg**
The tests are a bit ugly, but the cache otherwise hides the errors
away during testing.
The ternary operation did not work, because complex types do not like
assignment of 0.

Closes gh-11261

### CODE REVIEW

**Comment by charris** [comment]
Oops :)
```
  numpy/core/src/npysort/binsearch.c.src:49:9: warning: ???return??? with a value, in function returning void [enabled by default]
```

File: numpy/core/tests/test_multiarray.py [hunk 1]
```diff
@@ -2091,6 +2097,14 @@ def test_searchsorted_with_sorter(self):
             assert_equal(b, out)
             b = a.searchsorted(a, 'r', s)
             assert_equal(b, out + 1)
+            # Test empty array, use a fresh array to get warnings in
```
Comment (eric-wieser) [inline numpy/core/tests/test_multiarray.py]: What would happen if you constructed the empty array once here, and reused it?

Comment (eric-wieser) [reply numpy/core/tests/test_multiarray.py]: I've gone ahead and changed that - it makes the test substantially easier to read

**Review by eric-wieser — APPROVED** [review]

**Comment by charris** [comment]
Thanks Sebastian.

**Pull request merged**